Makefile

```diff
@@ -74,6 +74,41 @@ $(TOOLS)/gojq: PACKAGE=github.com/itchyny/gojq/cmd/gojq
 .PHONY: tools
 tools: $(CROSSLINK) $(DBOTCONF) $(GOLANGCI_LINT) $(MISSPELL) $(GOCOVMERGE) $(STRINGER) $(PORTO) $(GOJQ) $(SEMCONVGEN) $(MULTIMOD) $(SEMCONVKIT)
 
+# Virtualized python tools via docker
+
+# The directory where the virtual environment is created.
+VENVDIR := venv
+
+# The directory where the python tools are installed.
+PYTOOLS := $(VENVDIR)/bin
+
+# The pip executable in the virtual environment.
+PIP := $(PYTOOLS)/pip
+
+# The directory in the docker image where the current directory is mounted.
+WORKDIR := /workdir
+
+# The python image to use for the virtual environment.
+PYTHONIMAGE := python:3.11.3-slim-bullseye
+
+# Run the python image with the current directory mounted.
+DOCKERPY := docker run --rm -v "$(CURDIR):$(WORKDIR)" -w $(WORKDIR) $(PYTHONIMAGE)
+
+# Create a virtual environment for Python tools.
+$(PYTOOLS):
+# The `--upgrade` flag is needed to ensure that the virtual environment is
+# created with the latest pip version.
+	@$(DOCKERPY) bash -c "python3 -m venv $(VENVDIR) && $(PIP) install --upgrade pip"
+
+# Install a python package into the virtual environment.
+$(PYTOOLS)/%: | $(PYTOOLS)
+# The `--upgrade` flag is needed to ensure that the package is upgraded to the
+# latest version.
+	@$(DOCKERPY) $(PIP) install --upgrade $(PACKAGE)
```

Comment:
Do we want to pin the version of `codespell` or are we OK always using the latest?

The downsides of using the latest would be:
1. The build may suddenly start failing and interrupt other work
2. No control on what is exactly installed (so we have less control if `codespell` somehow does something harmful)

Same comment applies to the GitHub Action.

Maybe we could use `requirements.txt` and control the version there?
Also we could use the Make target instead of the GitHub Action as I do not see if it adds the possibility to configure `codespell` version. 

PS. It may be good to wait for @open-telemetry/go-maintainers for additional feedback if they agree with my comments or not.

Reply:
I do not think we are currently pinning any of the Go tools installed by Make, e.g. misspell, today, but I'm not opposed to pinning codespell to the version that it is currently installing now if that is what is being requested.

> The build may suddenly start failing and interrupt other work (my biggest concern)

The `codespell` target should not be failing any builds since it is not linked as a dependency in the `precommit` target like `misspell` is.

> Maybe we could use requirements.txt and control the version there?
Also we could use the Make target instead of the GitHub Action as I do not see if it adds the possibility to configure codespell version.

We should be able to pin the codespell version like so without the need for a requirements file:

```
(CODESPELL): PACKAGE=codespell==2.2.4
```

Reply:
All Go tools are pinned and updated via dependabot thanks to https://github.com/open-telemetry/opentelemetry-go/blob/main/internal/tools/go.mod

> We should be able to pin the codespell version like so without the need for a requirements file:

The downside of it is it will require manual work for updating the dependency.

Reply:
> All Go tools are pinned and updated via dependabot thanks to https://github.com/open-telemetry/opentelemetry-go/blob/main/internal/tools/go.mod

Ah, this is good to know. Thanks for the link.

> The downside of it is it will require manual work for updating the dependency.

Perhaps we can pin it for now and if the issue of upgrading the python dependencies becomes such a chore we can dedicate some engineering effort into automating it?

Reply:
> Perhaps we can pin it for now and if the issue of upgrading the python dependencies becomes such a chore we can dedicate some engineering effort into automating it?

I strongly prefer to integrate with dependabot as part of this PR. Otherwise nobody will upgrade the version.

Reply:
So are you recommending that I:

1. Update this PR to create a `requirements.txt` file and pin the version of codespell==2.2.4
2. Create another PR to update https://github.com/open-telemetry/opentelemetry-go-build-tools/tree/main/dbotconf to set up the dependabot configuration for pip
3. Create another PR to with the updated dbotconf to generate the `.github/dependabot.yml` file

?